src/features/session_replay/aggregate/index.js

```diff
@@ -152,46 +155,67 @@ export class Aggregate extends AggregateBase {
         raw: true
       }, this)
 
+      registerHandler('recordReplay', () => {
+        // if it has aborted or BCS returned bad entitlements, do not allow
+        if (this.blocked || !this.entitled) return
+        // if it isnt already (fully) initialized... initialize it
+        if (!recorder) this.initializeRecording(false, true, true)
+        // its been initialized and imported the recorder but its not recording (mode === off || error)
+        else if (this.mode !== MODE.FULL) this.switchToFull()
+        // if it gets all the way to here, that means a full session is already recording... do nothing
+      }, this.featureName, this.ee)
+
+      registerHandler('pauseReplay', () => {
+        this.actuallyStop(this.mode !== MODE.ERROR)
+      }, this.featureName, this.ee)
+
       // Wait for an error to be reported.  This currently is wrapped around the "Error" feature.  This is a feature-feature dependency.
       // This was to ensure that all errors, including those on the page before load and those handled with "noticeError" are accounted for. Needs evalulation
       registerHandler('errorAgg', (e) => {
         this.hasError = true
         this.errorNoticed = true
         // run once
         if (this.mode === MODE.ERROR && globalScope?.document.visibilityState === 'visible') {
-          this.mode = MODE.FULL
-          // if the error was noticed AFTER the recorder was already imported....
-          if (recorder && this.initialized) {
-            this.stopRecording()
-            this.startRecording()
-
-            this.scheduler.startTimer(this.harvestTimeSeconds)
-
-            this.syncWithSessionManager({ sessionReplayMode: this.mode })
-          }
+          this.switchToFull()
         }
       }, this.featureName, this.ee)
 
-      this.waitForFlags(['sr']).then(([flagOn]) => this.initializeRecording(
-        flagOn,
-        (Math.random() * 100) < getConfigurationValue(this.agentIdentifier, 'session_replay.error_sampling_rate'),
-        (Math.random() * 100) < getConfigurationValue(this.agentIdentifier, 'session_replay.sampling_rate')
-      )).then(() => sharedChannel.onReplayReady(this.mode)) // notify watchers that replay started with the mode
+      this.waitForFlags(['sr']).then(([flagOn]) => {
+        this.entitled = flagOn
+        this.initializeRecording(
+          (Math.random() * 100) < getConfigurationValue(this.agentIdentifier, 'session_replay.error_sampling_rate'),
+          (Math.random() * 100) < getConfigurationValue(this.agentIdentifier, 'session_replay.sampling_rate')
+        )
```

Comment:
Is it possible that this code could execute after a customer has called the API to start a recording? Any risk with this if they do?

Reply:
there is the chance this gets called after the api has executed, but there is a short-circuit in `initializeRecording` that prevents it from rerunning if already recording